### ISSUE TITLE
improve case with method in trait, one named one anonymous lifetime parameter

### ISSUE THREAD

**Issue by gaurikholkar** [rust-lang/rust#42702]
In #42669 , we are currently not improving the error message for the methods in traits. For this example,
```
trait Foo {
  fn foo<'a>(x: &i32, y: &'a i32) -> &'a i32 {
              //    -  consider changing the type of `x` to `&'a i32`
        if x > y { x } else { y }
             //      ^ - lifetime `'a` required 
  }
} 
note: ...changing the signature in the trait may require changing the corresponding impls of the trait
```
cc @nikomatsakis

**Comment by nikomatsakis** [comment]
Actually, I think we *do* cover this case in #42669 -- this would be a `NodeTraitItem`.

**Comment by nikomatsakis** [comment]
However, I think what I was concerned about here was that we might want to give *more* information (i.e., the `note` shown, although it seems a bit wordy). Not sure if that's worth it.

**Comment by steveklabnik** [comment]
Triage: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=af0e0b47620c290c33c3aacadc83e4db

```
error[E0621]: explicit lifetime required in the type of `x`
 --> src/lib.rs:4:13
  |
2 |     fn foo<'a>(x: &i32, y: &'a i32) -> &'a i32 {
  |                   ---- help: add explicit lifetime `'a` to the type of `x`: `&'a i32`
3 |         if x > y {
4 |             x
  |             ^ lifetime `'a` required

error: aborting due to previous error
```

I am not sure if this is good enough to close the bug or not.

**Comment by nikomatsakis** [comment]
I think we can close it. Thanks.
